DataFormats/Detectors/EMCAL/include/DataFormatsEMCAL/Cluster.h

```diff
@@ -0,0 +1,102 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#ifndef ALICEO2_EMCAL_CLUSTER_H_
+#define ALICEO2_EMCAL_CLUSTER_H_
+
+#include <array>
+#include <exception>
+#include <iosfwd>
+#include <string>
+#include <vector>
+#include "CommonDataFormat/TimeStamp.h"
+#include "CommonDataFormat/RangeReference.h"
+#include "MathUtils/Cartesian3D.h"
+
+namespace o2
+{
+
+namespace emcal
+{
+
+/// \class Cluster
+/// \brief EMCAL Cluster
+///
+class Cluster : public o2::dataformats::TimeStamp<Float16_t>, public Point3D<Float16_t>
```

Comment:
``Float16_t`` has precision ~10e-4, just to be sure: which time is stored with this precision?

Reply:
Hi @shahor02,

In principle we would prefer to store the calibrated time with respect to the bunch crossing and not the absolute time. In this case we would have a time window of 1.5 mus and a precision of ~1 ns, unit would be ns. For the desired precision we cannot go below a 16 bit data type.

Maybe this needs to be rediscussed, also with respect to what is needed to match EMCAL information with the continuous detectors. Eventually one can make the class templated itself?

Cheers

Markus

Reply:
@mfasDa In that case one will need to have BC identifier either on cluster on sub-container level. OK, we may come back to that later, will merge as it is once test are passed.

Reply:
@shahor02 We were expecting that there will be a common header from the system with BC and trigger information sent among components, similar to how this is made available to then components by the digitizer-workflow. Otherwise we also have a dedicated header for EMCAL https://github.com/AliceO2Group/AliceO2/blob/dev/DataFormats/Detectors/EMCAL/include/DataFormatsEMCAL/EMCALBlockHeader.h which could carry the information - currently it only signalises whether it has payload (exit condition for the reco workflow). In any case it can be rediscussed. It would however be helpful if the pull request in the current design can be merged as the development of the clusterizer, run2-3 convert and the QC depend on a prototype being merged.

Reply:
There will be certainly an interaction record associated with the primary vertex, we will probably equip it with a reference on the list of the relevant cell indices, to be decided later.